bittensor_cli/src/commands/root.py

```diff
@@ -569,14 +569,7 @@ async def get_stake_for_coldkey_and_hotkey(
         # Stake it all.
         staking_balance = Balance.from_tao(my_prev_coldkey_balance.tao)
```

Comment:
Hey! 

Since this also include undelegating, we should also make a distinction here when unstaking all:

```
    if amount is None:
        # Stake it all.
        if delegate_string == "delegate":
            staking_balance = Balance.from_tao(my_prev_coldkey_balance.tao)
        else:
            # Unstake all
            staking_balance = Balance.from_tao(my_prev_delegated_stake.tao)
    else:
        staking_balance = Balance.from_tao(amount)
```

Reply:
Fixed.

Unit tests need to be updated as well to check for the wallet balance change. Unit tests stopped working locally for me (infinite wait for the local subtensor). Once I resolve the issue I will update the tests too.

Reply:
Unit tests updated. 

However, I couldn't check the delegate balance when staking all. Float existential deposit, some bug or output value got truncated (rounded). Used wallet balance instead. Open for suggestions or could be just merged as is :)